src/main/java/com/eleks/academy/whoami/core/impl/TurnImpl.java

```diff
@@ -52,6 +53,29 @@ public List<List<PlayerAction>> getTurns() {
 		return this.turns; //TODO: history
 	}
 
+	@Override
+	public boolean isQuestionPresent() {
+		return this.getCurrentTurn().get(0).getValue() != null;
+	}
+
+	@Override
+	public boolean isAnswerer(String player) {
+		int index = this.getCurrentTurn()
+				.indexOf(new PlayerAction(player, PlayerAction.Action.ANSWER, null));
+		return index >= 0;
+	}
+
+	@Override
+	public boolean hasTurnEnded() {
+		int count = 0;
```

Comment:
easier is to return false in case first found value is `null` otherwise return true, no need to count

Reply:
I didn't notice that earlier.